### PR TITLE
fix https://github.com/libigl/libigl/issues/1343

### PR DESCRIPTION
**Pull request by alecjacobson**
floating point error could lead final value in cumsum to be <1. It should be ==1 so that the random values [0,1] do not exceed range spanned by cumsum.

(I couldn't manage to trigger a bad random value, but I confirmed that the old code had values of 0.99999999999993516 instead of 1.0. So I now believe it _could_ have happened.)

https://github.com/libigl/libigl/issues/1343

[Describe your changes and what you've already done to test it.]

#### Check all that apply (change to `[x]`)
- [ ] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [ ] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [x] This is a minor change.

### CODE REVIEW

File: include/igl/random_points_on_mesh.cpp [hunk 1]
```diff
@@ -35,6 +34,10 @@ IGL_INLINE void igl::random_points_on_mesh(
   A0.bottomRightCorner(A.size(),1) = A;
   // Even faster would be to use the "Alias Table Method"
   cumsum(A0,1,C);
+  const Scalar Cmax = C(C.size()-1);
+  assert(Cmax > 0 && "Total surface area should be positive");
+  // Why is this more accurate than `C /= C(C.size()-1)` ?
+  for(int i = 0;i<C.size();i++) { C(i) = C(i)/Cmax; }
```
Comment (jiangzhongshi) [inline include/igl/random_points_on_mesh.cpp]: I think the reason is: here the last value is explicitly made equal to `Cmax/Cmax` which will more likely be `1`.

Comment (alecjacobson) [reply include/igl/random_points_on_mesh.cpp]: I still don't get it. I expect it to be _exactly_ 1 in both cases. Why would Eigen's broadcaster not give the same result as this loop?

**Comment by jdumas** [comment]
Do you have an example of mesh/nb samples where the old code produced 0.99999999999993516 in the normalized cumsum?

File: include/igl/random_points_on_mesh.cpp (hunk 1 above)
Comment (jdumas) [reply include/igl/random_points_on_mesh.cpp]: I just tried on the meshes from the tutorial data, and I always get `1` if I replace this line by `C /= C(C.size()-1)`, even for a very large number of samples (10,000,000).

**Comment by alecjacobson** [comment]
I tried on the xyz dragon that has millions of tris

**Comment by jdumas** [comment]
So I tried also on this model, and if I use
```
  cumsum(A0,1,C);
  C /= C(C.size()-1);
  std::cout << std::hexfloat << C(C.size()-1) << std::endl;
```
Then it always prints exactly 1. So I don't really understand your comment _"Why is this more accurate than `C /= C(C.size()-1)` ?"_. To be fair, the old code (where you divide by the total area first `A /= A.array().sum();`) does produce a Cmax != 1, but that's not surprising since the floating point error accumulate in a different way when summing the normalized areas.

**Pull request merged**